examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -62,32 +66,76 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
     override fun onPurchasePackageClicked(cardView: View, currentPackage: Package, purchaseOption: PurchaseOption?) {
         binding.purchaseProgress.visibility = View.VISIBLE
 
-        if (purchaseOption == null) {
-            Purchases.sharedInstance.purchasePackageWith(
-                requireActivity(),
-                currentPackage,
-                { error, userCancelled ->
-                    if (!userCancelled) {
-                        showUserError(requireActivity(), error)
+        // TODO is this true? is it possible to have multiple subs to one app at once,
+        //  in which case we don't do an update but simply purchase again?
+        val isUpgrade = currentPackage.product.type == ProductType.SUBS && activeSubscriptions.isNotEmpty()
```

Comment:
this logic is something the developer could handle (whether something is an upgrade or not), i think? we might want to think through what has changed with BC5...

this is the point i realized we might need the purchaseoption available in customer info... there's no way to know which base plan was purchased. the developer then can't disable/enable purchasing the same baseplan

see convo [here](https://revenuecat.slack.com/archives/C03M70X8DGF/p1671642971144009)

Reply:
To me it seems that we should extend the CustomerInfo API and include the subscription and baseplanid in the product info.
Maybe we can add new fields that would include the baseplans. In the end that's what identifies baseplans now, so we shoudl include it.

So all these maps would have to change from a String to the tuple (unless we decide to use the concatenated version).
https://github.com/RevenueCat/purchases-android/blob/main/public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt#L43

This would also obviously require changes in the backend in the subscriber serializer

Reply:
I think I might need some clarification here 🤔 I'm more familiar with the App Store side where an upgrade/downgrade is done by Apple when a subscription is purchased in the same subscription group. The new model does have something similar to these were the subscription is the group and the base plans are the subscriptions that can be upgrade/downgraded... and only on base plan can be purchase at once.

It did some testing and it appears that Google enforces that one one of those base plans can be active at once so we _should_ be able to this `isUpgrade` if it the old and new are in the same subscription group thingy.

However, it seems like that a developer could also off the upgrade/downgrade option between two separate subscriptions. 👈 **Is my understanding here correct?** 

If my understanding is correct, it _seems_ like to make this work for ever use case, we could do this `isUpgrade` logic but we _should_ definitely need to allow the developer to pass in if its an upgrade/downgrade (especially if they are coming from a subscription setup before BC5 / base plans and want to keep using there separate subscriptions for different tiers.

Are my thoughts here correct? 😇

Reply:
OMG... I just realized that this was in Purchase Tester and not in our SDK 🤦‍♂️ 

I _think_ most of my above comment could be ignored 😇 This logic _does_ feel kind of advanced even though. I'm not sure if we should either:
- provide a helper to for this logic for the developer
- provide an example of this in the docs (https://www.revenuecat.com/docs/managing-subscriptions) on ways to handle this

Even though is in the example app, I don't want to make developers come here to see this. I think showing how to handle upgrade/downgrade might be a nice docs addition.

Reply:
>  I don't want to make developers come here to see this. I think showing how to handle upgrade/downgrade might be a nice docs addition.

totally agree. actually, purchase-tester isn't meant to be somewhere for our users to look anyway, it's more for internal usage. we should add a cleaner example to `MagicWeather` once we have this locked down.

but either way, there is an opportunity to update docs or even to create helper methods in the SDK.

Reply:
@swehner i think having base plan exposed is helpful outside of the upgrade/downgrade case too, what do you think? i can imagine wanting to display which duration a user is subscribed to somewhere in the app... 

we should investigate the effort that'd take on backend..

Reply:
Yes, that makes a lot of sense - I think the basePlan needs to be exposed for sure. Duration is something we can add if needed. I've changed the ticket to implementing rather than investigating. We should go for it!